centreon/src/Core/Command/Infrastructure/Repository/ApiWriteCommandRepository.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+/*
+ * Copyright 2005 - 2023 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types = 1);
+
+namespace Core\Command\Infrastructure\Repository;
+
+use Centreon\Domain\Log\LoggerTrait;
+use Core\Command\Application\Repository\WriteCommandRepositoryInterface;
+use Core\Command\Domain\Model\Argument;
+use Core\Command\Domain\Model\NewCommand;
+use Core\Command\Infrastructure\Model\CommandTypeConverter;
+use Core\CommandMacro\Domain\Model\NewCommandMacro;
+use Core\Common\Infrastructure\Repository\RepositoryTrait;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+class ApiWriteCommandRepository implements WriteCommandRepositoryInterface
+{
+    use LoggerTrait;
+    use RepositoryTrait;
+
+    private ?string $proxy = null;
+
+    private ?string $url = null;
+
+    private string $authenticationToken = '';
+
+    private int $timeout = 60; // Default timeout
+
+    public function __construct(readonly private HttpClientInterface $httpClient)
+    {
+    }
+
+    public function setProxy(string $proxy): void
+    {
+        $this->proxy = $proxy;
+    }
+
+    public function setUrl(string $url): void
+    {
+        $this->url = rtrim($url, DIRECTORY_SEPARATOR);
+        if (! str_starts_with($this->url, 'http')) {
+            $this->url = 'https://' . $this->url;
+        }
+    }
+
+    public function setAuthenticationToken(string $token): void
+    {
+        $this->authenticationToken = $token;
+    }
+
+    public function setApiTimeOut(int $timeout): void
+    {
+        $this->timeout = $timeout;
+    }
```

Comment:
As this code is duplicated in the apiWriteMediaRepository and if more migrations useCase will be created, I think it should be great to avec an AbstractApiRepository which contains those setters

Reply:
👍 I might be a good idea, I take note for next command